### DISCUSSION TITLE
No hashes generated?

### DISCUSSION THREAD

**Discussion by Lani** [RockiRider/csp#54] [Q&A] [answered]
Hi,

This seems like a really good and promising plugin. But I think that I am missing something, I expected this plugin to generate integrity hashes from the bundles and add them to index.html script tags and also as CSP. But I cannot get that to work, either locally, or by cloning the repo and building the React example.

I also tried setting up a new project in Stackblitz and add the plugin there, but no hashes in the index.html file after the project is built.

https://stackblitz.com/edit/vitejs-vite-ku4wkr

**Comment by tsotimus** [comment, marked as answer]
Hey @Lani!

Totally missed this sorry for the late reply!

### Current state of the plugin
Automatically generates hashes for you in dev mode, because otherwise your CSP just breaks

[Opt in for build](https://vite-csp.tsotne.co.uk/api-docs#build)
- The reason being you don't need hashing to have a solid CSP, because everything is built to an index.js and index.css, so just having 'self' is good enough in prod
- SRI(https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) is typically used for external resources, although you could argue its just good to have
- Currently the plugin does support SRI, but only partially. If you implement any lazy loading (have multiple js bundles outputted at build time) SRI won't work. And even if it did, adding an integrity hash to the lazy loaded scripts is difficult

### Future

I am looking at doing a breaking change over December to introduce SRI properly under a better API than `build.hash`, will probably be `build.sri` and depending on if I am able to deal with the lazy loading issue it could be defaulted to true.


Hope this helps! Its been a few months since I've worked on the SRI feature, so feel free to challenge anything I've said

**Reply by Lani** [reply]
I started to realize that myself soon after posting this. But it was great to get a reply from you to verify my own understanding, thanks 🙂
